oracle/blockeligibilityvalidator.go

```diff
@@ -55,15 +58,28 @@ func (v BlockEligibilityValidator) BlockEligible(block *types.BlockHeader) (bool
 	epochBeacon := v.beaconProvider.GetBeacon(epochNumber)
 	message := serializeVRFMessage(epochBeacon, epochNumber, counter)
 	vrfSig := block.EligibilityProof.Sig
-	res, err := v.validateVRF(message, vrfSig, []byte(block.MinerID.VRFPublicKey))
+
+	pubKey, err := ed25519.ExtractPublicKey(block.Bytes(), block.Sig())
```

Comment:
pubnlic key extraction code is duplicated from caller function `confirmBlockValidity`, see if you can refactor to avoid this duplication

Reply:
cool, i moved all the logic to BlockEligible so the extraction happens once